pallets/hsm/src/lib.rs

```diff
@@ -785,7 +788,9 @@ pub mod pallet {
 			if hollar_amount_to_trade > 0 {
 				let hsm_account = Self::account_id();
 
-				Self::mint_hollar(&hsm_account, hollar_amount_to_trade)?;
+				// If flash minter not set, we use HSM facilitator.
+				let flash_minter = T::FlashMintFacilitator::get();
+				Self::mint_hollar(&hsm_account, hollar_amount_to_trade, flash_minter)?;
```

Comment:
this is not how flashloan works, you have to wrap you call into it, flashloan itself makes sure that the loan is burned in the end

see impl https://github.com/aave/gho-core/blob/397336cc1cc6d27a28a82564443a77ba049d3a48/src/contracts/facilitators/flashMinter/GhoFlashMinter.sol#L68